evdi broken: /build/source/module/evdi_drv.h:23:10: fatal error: linux/reservation.h: No such file or directory

## Issue description

Tried to update my unstable installation and it fails on evdi compliation:

```
'/nix/store/j5gbwl29jg11sgc8cjk2h8v95sq06fiy-busybox-1.31.1/sbin/cut' -> '/nix/store/jfc463n33x86wn3ylclag0gwyx80mz7z-extra-utils/bin/cut'
building '/nix/store/2iywkkpvirmmj0sn2xlb3b2ji9mmkzjk-fc-cache.drv'...
In file included from /build/source/module/evdi_drv.c:16:
/build/source/module/evdi_drv.h:23:10: fatal error: linux/reservation.h: No such file or directory
 #include <linux/reservation.h>
          ^~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

## Technical details

Please run `nix run nixpkgs.nix-info -c nix-info -m` and paste the result.

 - system: `"x86_64-linux"`
 - host os: `Linux 5.3.11, NixOS, 20.03pre202088.e89b21504f3 (Markhor)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.1`
 - channels(hyper): `"nixos-19.09pre185402.5eac339829b"`
 - channels(root): `"nixos-20.03pre203172.0ee0489d42e"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

--- a/pkgs/os-specific/linux/evdi/default.nix
+++ b/pkgs/os-specific/linux/evdi/default.nix
@@ -1,37 +1,44 @@
-{ stdenv, fetchFromGitHub, kernel, libdrm }:
+{ stdenv, fetchFromGitHub, fetchpatch, kernel, libdrm }:
 
 stdenv.mkDerivation rec {
   pname = "evdi";
-  version = "1.6.4";
+  version = "-unstable-20190116";
 
   src = fetchFromGitHub {
     owner = "DisplayLink";
     repo = pname;
-    rev = "v${version}";
-    sha256 = "1yrjm8lvvz3v4h5af6m9qzq6z4lbgd7qbvq5rz7sjhdsh7g6qibd";
+    rev = "391f1f71e4c86fc18de27947c78e02b5e3e9f128";
+    sha256 = "147cwmk57ldchvzr06lila6av7jvcdggs9jgifqscklp9x6dc4ny";
   };
 
   nativeBuildInputs = kernel.moduleBuildDependencies;
 
   buildInputs = [ kernel libdrm ];
 
+  patches = [
+    (fetchpatch {
+      url    = "https://crazy.dev.frugalware.org/evdi-all-in-one-fixes.patch";
+      sha256 = "03hs68v8c2akf8a4rc02m15fzyp14ay70rcx8kwg2y98qkqh7w30";
+    })
+  ];
+
   makeFlags = [
-    "KDIR=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build"
     "KVER=${kernel.modDirVersion}"
+    "KDIR=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build"
   ];
 
   hardeningDisable = [ "format" "pic" "fortify" ];
 
   installPhase = ''
     install -Dm755 module/evdi.ko $out/lib/modules/${kernel.modDirVersion}/kernel/drivers/gpu/drm/evdi/evdi.ko
-    install -Dm755 library/libevdi.so $out/lib/libevdi.so
+    install -Dm755 library/libevdi.so.1.6.4 $out/lib/libevdi.so
   '';
 
   meta = with stdenv.lib; {
     description = "Extensible Virtual Display Interface";
-    homepage = "https://www.displaylink.com/";
-    license = with licenses; [ lgpl21 gpl2 ];
     platforms = platforms.linux;
+    license = with licenses; [ lgpl21 gpl2 ];
+    homepage = "https://www.displaylink.com/";
     broken = versionOlder kernel.version "4.9" || stdenv.isAarch64;
   };
 }

